src/python/pants/util/dirutil.py

```diff
@@ -66,7 +67,15 @@ def safe_mkdir(directory, clean=False):
   if clean:
     safe_rmtree(directory)
   try:
-    os.makedirs(directory)
+    if directory.endswith(get_repo_root() + '/.pants.d'):
+      # make sure the base directory for .pants.d(symlink) exists
+      if not os.path.isdir('/tmp/.pants.d'):
```

Comment:
os.path.isdir may not be a good proxy for whether or not *something* exists at that path. .pants.d for instance could be a file.

Reply:
Added both isfile and isdir check, thanks.